pkg/signup/service/signup_service.go

```diff
@@ -438,7 +438,8 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, provider ResourceProvider, u
 	// Check UserSignup status to determine whether user signup is complete
 	approvedCondition, approvedFound := condition.FindConditionByType(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupApproved)
 	completeCondition, completeFound := condition.FindConditionByType(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
-	if !approvedFound || !completeFound || approvedCondition.Status != apiv1.ConditionTrue {
+	if !approvedFound || !completeFound ||
+		( approvedCondition.Status != apiv1.ConditionTrue && approvedCondition.Reason != toolchainv1alpha1.UserSignupUserDeactivatedReason) {
```

Comment:
I think we should replace it by:
```suggestion
		condition.IsFalseWithReason(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupApproved, toolchainv1alpha1.UserSignupPendingApprovalReason {
```
This is the actual condition we set for UserSignup when it's pending approval:
```
  - reason: PendingApproval
    status: "False"
    type: Approved
```

Reply:
I can do that if you prefer but my concern would be that is is more strict than the current code which could lead to other behavior changes.

Reply:
Yeah, I still think it's better to rely on the `PendingApproval` reason.

Reply:
Updated